src/snowflake/snowpark/mock/_functions.py

```diff
@@ -1085,6 +1085,43 @@ def mock_to_variant(expr: ColumnEmulator):
     return res
 
 
+def _object_construct(exprs, drop_nulls):
+    import pandas
+
+    expr_count = len(exprs)
+    if expr_count % 2 != 0:
```

Comment:
There is also the case when an asterisk is used where the object is constructed using the attribute names as keys and the associated tuple values as values.

Example:
```
SELECT province, created_date
    FROM demo_table_1
    ORDER BY province;
+----------+--------------+
| PROVINCE | CREATED_DATE |
|----------+--------------|
| Alberta  | 2020-01-19   |
| Manitoba | 2020-01-18   |
+----------+--------------+

SELECT OBJECT_CONSTRUCT(*) AS oc
    FROM demo_table_1
    ORDER BY oc['PROVINCE'];
+---------------------------------+
| OC                              |
|---------------------------------|
| {                               |
|   "CREATED_DATE": "2020-01-19", |
|   "PROVINCE": "Alberta"         |
| }                               |
| {                               |
|   "CREATED_DATE": "2020-01-18", |
|   "PROVINCE": "Manitoba"        |
| }                               |
+---------------------------------+
```
where there could an odd number of columns in `expr`.  This might require some logic in `_plan.py` since we need to distinguish the case when the child expression is `*` instead of explicitly selecting all columns.

Reply:
I've created SNOW-1292493 to track adding that functionality.